Support: Introduce Help Flow

Aurorum opened the pull request:
_**This PR has heavily changed since writing this description, please see a more updated one [here](https://github.com/Automattic/wp-calypso/pull/35944#issuecomment-528104262)**_

#### Changes proposed in this Pull Request

Okay, this is a kinda ambitious fix for #16016 by adding an extra step to the Support flow. This step ensures the user doesn't access the contact form if they wouldn't be able to get help with that form anyway.

<img width="877" alt="Screenshot 2019-09-03 at 13 52 27" src="https://user-images.githubusercontent.com/43215253/64177947-6779f300-ce58-11e9-92e2-df21c018b1f0.png">

Each button is displayed depending on the user, so:

- WordPress.com: always displays (redirects to contact form)
- Jetpack: displays if the user has a Jetpack site (redirects to https://jetpack.com/contact-support/)
- WordAds: displays if the user has a site with a paid plan (redirects to contact form)
- WooCommerce: displays if the user has a Jetpack or Atomic site

Clicking each button displays a dialog when support can't be offered using the Support Form on WordPress.com. For example:

<img width="967" alt="Screenshot 2019-09-03 at 13 52 54" src="https://user-images.githubusercontent.com/43215253/64178872-1f5bd000-ce5a-11e9-89a9-fdfeaf74d752.png">

An "Issue with my blog" is a bit more complicated. If the user has WordPress.com sites **and** Jetpack sites, this should appear.

If the user's site isn't listed and they click the link, they'll be redirected to the contact form. Otherwise, there's a prompt to visit the self-hosted forums, where forum members usually end up redirecting them.

<img width="903" alt="Screenshot 2019-09-03 at 13 52 40" src="https://user-images.githubusercontent.com/43215253/64178129-c9d2f380-ce58-11e9-99ea-ba8eec01033f.png">

Sometimes though, this dialog doesn't display. This happens in these events:

- The user has no sites 
- All the sites the user has are simple WordPress.com sites

In such scenarios, I believe that the Contact Form is the most appropriate place for them to go, so they are immediately directed there. 

#### Advantages to this Approach

I did consider several options before this, such as displaying a notice on the Contact Form, but ultimately, this was the best I could come up with. There's several reasons why:

- Sometimes flows can get complicated, and one notice isn't adequate. For example, I have several simple sites, Jetpack sites, WordAds and an atomic site. It'd be bizarre displaying warnings for all of them 
- This approach allows any warnings to be ignored by people who don't need to see them (since the Help Flow is skipped in the cases above) 
- It's also possible to provide a direct link to the form where that would be useful (I can imagine some scenarios, such as Happiness providing them)

#### Reviews 

@KokkieH, is there anything else I'm missing where support in the forums can't be provided? Would it make sense, for example, for there to be another button for WordPress apps? Do you prefer this solution to the current flow, and do you think it'd resolve the issue? 

I'm not exactly sure who to ask to review this. @donpark @sixhours, is it possible for this to please be reviewed under `FixtheFlows`, since this is all about flows after all!

**Copy:**
(cc @michelleweber, would you mind taking a look?) 

Screenshot for context (mobile view btw, desktop view above): 
<img width="445" alt="Screenshot 2019-09-03 at 14 52 25" src="https://user-images.githubusercontent.com/43215253/64179091-75c90e80-ce5a-11e9-884d-e0a65df09129.png">
> Hi there! How can we help?
> Please select the service which you're having issues with, so you can be directed to the ideal place for support.
> Please select the service which you're having issues with, so you can be directed to the ideal place for support.
> An issue with my blog

If you're unsure of the service you require assistance with, click here.

Screenshot for context: 
<img width="512" alt="Screenshot 2019-09-03 at 14 00 47" src="https://user-images.githubusercontent.com/43215253/64178908-297dce80-ce5a-11e9-818b-25a9976f524b.png">

> We wish we could help more...
> Please contact the Jetpack support team directly for issues regarding the plugin, or explore the Jetpack support documentation.
> Please contact the WooCommerce support team directly for issues regarding the Store service or the plugin.


Screenshot for context: 
<img width="903" alt="Screenshot 2019-09-03 at 13 52 40" src="https://user-images.githubusercontent.com/43215253/64178850-1539d180-ce5a-11e9-8dcc-94b681560c78.png">

> Which blog are you having issues with?
> Are you experencing an issue with one of the sites listed? If so, as your site is not hosted on WordPress.com, you should instead visit the self-hosted forums, or alternatively contact your hosting provider.
> Explore self-hosted site support
> The site I'm having issues with isn't listed

I'm welcome to alternative solutions to this issue, but this really feels like the best way forward!

Fixes #16016

Aurorum committed:
Verify Back route

Aurorum committed:
Create HelpFlow

a8ci18n commented on client/me/help/help-flow/index.jsx:
```diff
@@ -0,0 +1,226 @@
+
+		if ( jetpackDialog ) {
+			buttonLink = 'https://jetpack.com/contact-support/';
+			buttonText = translate( 'Contact Jetpack support' );
```
Hi! I've found a possible matching string that has already been translated **36** times:
`translate( 'Contact Jetpack Support.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Aurorum committed:
Hopefully turn tests green

Aurorum commented:
Thanks @sixhours! :) 

Managed to turn the tests green, so this should be ready for review now.

michelleweber commented:
Some updates for the copy, and a few questions/notes:

> Hi! How can we help?
> Tell us which service are you having issues with, and we'll send you right to the best place for support.
>
> If you're not sure what the problem is, click here.
> 

Questions on this on: 
- It feels like there's insufficient differentiation between "WordPress.com," "Jetpack," and "An issue with my blog." If I have a blog on WordPress.com, how do I know which one to select? What about if I have a self-hosted blog with Jetpack?
- What happens if I click "click here"? I'd love to be specific about where that goes instead of just saying "click here."

> We wish we could help more...
> For issues with the Jetpack plugin, contact the Jetpack support team directly or take a look at the Jetpack support documentation.

> For Store or WooCommerce questions, contact the Woo support team directly.

The screenshot here doesn't match the text, so just want to make sure this is the right copy.

> Which blog are you having issues with?
> Are you having problems with one of the sites listed here? These site aren't hosted on WordPress.com, so you'll visit the support forms for self-hosted site or contact your hosting provider.
> Explore self-hosted site support
> The site I'm having issues with isn't listed

Aurorum commented:
Thanks @michelleweber, I'll update the copy soon! :)

> What happens if I click "click here"? I'd love to be specific about where that goes instead of just saying "click here."

It redirects to the contact form where a user can send a message to Happiness.

The reasoning for this is that my intention is not to block people from accessing this form, only to make it easier for people to access the right place to get support for their specific issue. If they're unsure, it's probably best to just let them contact a human.

<img width="790" alt="Screenshot 2019-09-03 at 19 51 20" src="https://user-images.githubusercontent.com/43215253/64200339-36afb300-ce84-11e9-941e-d1a5cd5e4774.png">

#

> It feels like there's insufficient differentiation between "WordPress.com," "Jetpack," and "An issue with my blog." If I have a blog on WordPress.com, how do I know which one to select? What about if I have a self-hosted blog with Jetpack?

The intention behind the WordPress.com button was for issues with WordPress.com itself, for example a user having issues with the editor. That's similar for the Jetpack button.  For "An issue with my blog", the intention is that it displays a dialog with a list of self-hosted sites. 

The reasoning the "An issue with my blog" is necessary though is in case the user has no WordPress.com sites and only a Jetpack site, but has an issue with something on WordPress.com (like something being broken, for example). It's also beneficial for users who aren't certain whether their site is self-hosted or might not understand what that means, as the dialog displays a list of self-hosted sites.

I agree that can be confusing though, and I wonder if adding a tooltip to clarify the difference would be beneficial.

#

> The screenshot here doesn't match the text, so just want to make sure this is the right copy.

Yes, sorry, you're right.

<img width="512" alt="Screenshot 2019-09-03 at 14 00 47" src="https://user-images.githubusercontent.com/43215253/64201194-fe10d900-ce85-11e9-859b-71dbe400cacc.png">

> We wish we could help more...

> You don't have a site hosted at WordPress.com! If you're having an issue with a particular theme or plugin, contact its developer. Otherwise, explore the self-hosted forums for community help, or contact your hosting provider.

#
The copies you suggested apply to this screen below, and work well in context, thanks!

<img width="716" alt="Screenshot 2019-09-03 at 20 05 23" src="https://user-images.githubusercontent.com/43215253/64201267-2d274a80-ce86-11e9-9555-02d06000b83d.png">

GaryBairead commented:
If a user has a self-hosted site, and clicks the "WordPress.com button", would they be redirected to the contact form to create a thread in the dotcom support forums?

Aurorum commented:
Yes

GaryBairead commented:
> Yes

In that scenario, I'm not sure the current wording would address https://github.com/Automattic/wp-calypso/issues/16016

The core problem is that people aren't aware of the distinction between dotcom and dotorg, leading them to seek assistance for self-hosted sites in the dotcom forums.

Asking them to select between needing assistance for "WordPress.com" and "an issue with my blog" doesn't make that difference explicit - a WordPress.com site can be a blog, a self-hosted site doesn't have to be a blog.

I suspect most of [the people who erroneously create threads for dotorg sites](https://en.forums.wordpress.com/topic-tag/org) would simply click the WordPress.com button because they need help for "WordPress" in general.

Perhaps asking if they need assistance for a site hosted with WordPress.com or a self-hosted WordPress site hosted with a different provider would make the difference clearer.

Aurorum commented:
I understand. 

I think the confusion in this conversation alone reflects that some more clarity is needed, so I'm going to remove the "An issue with my blog" button in favour of something equally understandable, but more clear. 

I'll probably push those changes tomorrow. :)

donpark commented:
My feedback:

1. I like the general direction.
2. Design needs more work and feedback.
3. I think it's too large to be a FixTheFlows patch.

KokkieH commented:
So much has changed in support since I created the original issue for this, and the lines of where to get support has become more blurred, not less:  With Atomic, WordPress.com support now helps with both Jetpack and WooCommerce issues in some cases, and for some self-hosted site issues, specifically Automattic themes that don't have their own forums on WordPress.org, the dotcom forums are the best place for support. 

WordAds don't only rely on having a paid plan, as Jetpack users and legacy WordPress.com domain owners can also apply and be approved for WordAds without having a plan. 

We get occasional support requests for WooCommerce via this contact from people who have neither a Jetpack-connected nor an Atomic site, so only showing the WooCommerce option under those conditions will cause people who need it to miss it.

And personally I don't like the idea of this form taking any user to a dead end that says "Sorry, we can't help", and giving only the options to go elsewhere or close the request. I feel someone should always have the option to still submit their question, even if they're not sure if they're in the right place. Ultimately having a human point you to an alternative is a better experience than an automated brush-off :)

Also keep in mind that since the original issue was opened we've introduced the inline help button in Calypso as a more discoverable way than the contact form to open a support request, and for any changes we make to the form we should also consider if that should be incorporated in inline help, and what effect that might have as well.

I love the idea of intelligently routing people to the best place for them to get help, if we have the data needed to do so. But I agree with @donpark that this is perhaps a bigger project than can be addressed in a single PR, and besides design feedback we might need to first also step back and check how the needs in terms of routing support requests to the right channels have changed in the past two years.

supernovia commented:
I really like the direction. We have a similar signpost type feature at the [logged out contact page](https://en.support.wordpress.com/contact/) too.

![Screen Shot 2019-09-04 at 9 57 32 AM](https://user-images.githubusercontent.com/3476361/64271455-95177880-cefa-11e9-92f0-0999c20c3407.png)

Aurorum committed:
Push some changes

Aurorum committed:
Update styling

a8ci18n commented on client/me/help/help-flow/index.jsx:
```diff
@@ -0,0 +1,243 @@
+							</Button>
+							{ jetpackDialog && (
+								<Button primary onClick={ this.handleDialogClosure }>
+									{ translate( 'Contact Jetpack Support' ) }
```
Hi! I've found a possible matching string that has already been translated **36** times:
`translate( 'Contact Jetpack Support.' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Aurorum commented:
Thanks all for the review! There's still a few things I need to address:

- checking some more flows properly
- figuring out something for themes
- addressing lint errors

but that's for tomorrow, and I wanted to commit some changes anyway to show the general direction that's changed, and gather some thoughts

# 

The main concern seemed to be that the "An issue with my blog" button wasn't clear. I've removed that now in favour of this screen. It's intended to be easy to understand by:

- displaying a list of Jetpack connected blogs if the user also has WordPress.com blogs
- using the Jetpack or WordPress.com colours which users would be familiar with (especially Jetpack users)

<img width="835" alt="Screenshot 2019-09-04 at 22 31 33" src="https://user-images.githubusercontent.com/43215253/64293283-28d15f00-cf64-11e9-81c0-c7f2ea150c24.png">

This shouldn't take away from the other issues though. The "Other" button opens up this dialog, with expandable accordions:

<img width="913" alt="Screenshot 2019-09-04 at 22 31 27" src="https://user-images.githubusercontent.com/43215253/64293450-8665ab80-cf64-11e9-9a61-48952407bb64.png">

There's no logic on what's hidden and what's not, based on @KokkieH's comment. Clicking "My Jetpack-connected site" opens this screen:

<img width="898" alt="Screenshot 2019-09-04 at 22 31 38" src="https://user-images.githubusercontent.com/43215253/64293504-a09f8980-cf64-11e9-9a01-9391d61f97fa.png">

I'd appreciate if anyone could confirm whether or not this is less confusing 

# 

@KokkieH, thanks for the detailed response!

> With Atomic, WordPress.com support now helps with both Jetpack and WooCommerce issues in some cases, and for some self-hosted site issues, specifically Automattic themes that don't have their own forums on WordPress.org, the dotcom forums are the best place for support.

I'm going to take a closer look at spinning something up for themes tomorrow. I fear though that what's available might be restrictive for an ideal flow, but I've not taken a proper look yet. 

> We get occasional support requests for WooCommerce via this contact from people who have neither a Jetpack-connected nor an Atomic site, so only showing the WooCommerce option under those conditions will cause people who need it to miss it.

I wasn't aware of this, I've removed any logic for the new dialog so this should no longer be an issue.

> And personally I don't like the idea of this form taking any user to a dead end that says "Sorry, we can't help", and giving only the options to go elsewhere or close the request. I feel someone should always have the option to still submit their question, even if they're not sure if they're in the right place. Ultimately having a human point you to an alternative is a better experience than an automated brush-off :)

I think this is a tricky balance to ensure that a user doesn't feel more annoyed submitting their question to the wrong place and then having to wait to be redirected to the correct place. I also agree with you regarding users having the option to submit a form.

I've tried to rework this into sounding more like "Hey, we think you'd get better help somewhere else, but feel free to contact us if you're unsure" rather than "we can't help here" by changing the language and adding a link to open the contact form at the bottom left.

<img width="898" alt="Screenshot 2019-09-04 at 22 31 38" src="https://user-images.githubusercontent.com/43215253/64293504-a09f8980-cf64-11e9-9a01-9391d61f97fa.png">

> But I agree with @donpark that this is perhaps a bigger project than can be addressed in a single PR

The root issue of users submitting their queries/issues to the wrong place won't be completely solved by this PR alone, though I think that the `/help/contact` flow is the main flow causing this problem. There are other things that need to be addressed too, like the Inline Help issue you mentioned, but that's probably for another day.

# 

@michelleweber Sorry, I'm a bit hesitant to ask for a copy check immediately because this could be completely changed. Just in case though, here's the new copy:

<img width="913" alt="Screenshot 2019-09-04 at 22 31 27" src="https://user-images.githubusercontent.com/43215253/64294032-ce390280-cf65-11e9-903e-e66f940edf60.png">

> How can we help?
> **I have an issue with a custom theme/plugin**
> For issues with a custom theme or a custom plugin, contact that developer directly.
> **I have an issue with the WordPress app**
> For issues with the WordPress apps, use the in-help support located under "Help & Support" to provide additional information necessary for debugging the issue.
> **I have an issue with WordAds**
> For issues with WordAds, contact us here.
> **I have an issue with my Dashboard on WordPress.com**
> For issues with your WordPress.com dashboard, contact us here.
> I'd like to discuss something else and speak to a human

<img width="840" alt="Screenshot 2019-09-04 at 22 49 24" src="https://user-images.githubusercontent.com/43215253/64294270-4273a600-cf66-11e9-9dbc-f9fed495861d.png">

> We think we could provide more help elsewhere…
> I'd like to speak with a human anyway

<img width="835" alt="Screenshot 2019-09-04 at 22 31 33" src="https://user-images.githubusercontent.com/43215253/64294302-56b7a300-cf66-11e9-8ea2-e615270ba343.png">

> Hi there! How can we help?
> Please select the service which you're having issues with, so you can be directed to the ideal place for support.
> If you're unsure of the service you require assistance with, click here.

# 

Although there's still some more changes to be made, I'm curious as to if folks feel this is a more ideal flow, especially compared to the previous confusing one.

KokkieH commented:
This new flow feels much better to me. Giving only two choices in the first step makes things much simpler, and I think you cover the most common other issues in the second list.

One small nitpick: An issue with "My Jetpack Site" will not necessarily be a Jetpack issue, so it might be better to make the first split based on "My WordPress.com Site" and "The Jetpack Plugin", so we make a distinction between site issues and plugin issues - Jetpack support can only help with the latter.

Another small nitpick:

> I'd like to discuss something else and speak to a human

"speak to a human" could be interpreted as chat or phone support. I'd rather go with language like:

> Something else/not sure? Open a support request with WordPress.com, and if we can't help, we'll point you in the right direction for support.

KokkieH commented:
Another option we could include is unknown charges/billing issues - go to wpchrg.wordpress.com

Also missing, which I think should be there, is an option to direct people to WordPress.org/support:

**I have an issue with a WordPress site hosted somewhere else**
For help with the open source WordPress software, see https://wordpress.org/support/, or contact your hosting provider directly.


We also get the occasional Akismet and VaultPress issue submitted, usually billing-related, but those happen rarely enough that I don't think it's necessary to include separate options for them.

Aurorum committed:
Bunch of changes

Aurorum committed:
Fix design on smaller screens

Aurorum commented:
Thanks @KokkieH! I've pushed two more commits with the following:

- Rename the Jetpack button to "The Jetpack Plugin"
- Switch the "speak to a human wording"
- Includes billing and having a self-hosted site as issues in the "Other" screen
- Fixed the design on mobile
- Addressed lint errors

<img width="833" alt="Screenshot 2019-09-05 at 20 14 50" src="https://user-images.githubusercontent.com/43215253/64373111-727c8100-d01a-11e9-8243-6111305fed0d.png">

Unfortunately, one of the things which I was not able to do was to come up with a better flow for themes. I suspect the ideal flow would involve checking if the user has any themes which aren't created or maintained by Automattic, but I don't think it's possible right now. 

It would be nice to be able to use all available information to provide the ideal flow as initially intended, but that has several faults: there isn't enough information, it can become confusing, and sometimes that can cause less help instead, as @KokkieH mentioned.

As such, I feel like this is a good place to start with this PR, as it does help address the main issues of providing the user the best route for help. 

So, I think that this is ready for a proper review, and I'd appreciate confirmation that this PR no longer causes the confusion which it did initially. Any and all thoughts appreciated. :)

a8ci18n commented on client/me/help/help-flow/index.jsx:
```diff
@@ -0,0 +1,279 @@
+								primary
+								onClick={ this.handleJetpackButtonClick }
+							>
+								{ translate( 'The Jetpack Plugin' ) }
```
Hi! I've found a possible matching string that has already been translated **25** times:
`translate( 'Install the Jetpack plugin.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

bluefuton commented:
Hi @Aurorum! Sorry for the delay on this one. There are some solid improvements here and we'd love to get them merged. 

Please could you rebase the branch against trunk for us and we'll give it another review?

Aurorum committed:
Merge branch 'trunk' into add/help-flow

Aurorum committed:
Update code

Aurorum commented:
I must admit, I'm not sure what direction would be preferable with this PR. I've committed something which was as far as we got in 2019, but I'm happy to make any changes!

sixhours commented:
👋 I brushed off the dust and booted this up! :D I recorded a quick video walkthrough to give folks a sense of the current state.

https://user-images.githubusercontent.com/2124984/120030633-b7ea9a00-bfc5-11eb-830b-ced14fb62469.mov

I'd like to get our Happiness folks to take a look at this with fresh eyes and see if it's still something that would improve their lives. If so, I'll take point on reviewing and testing so we can get this shipped!

/cc @KokkieH since you were very helpful throughout this process and @jordesign since you're one of my pod people and might know who else to ask. :D

Aurorum commented:
If I'm entirely honest, I'm not too keen on this approach anymore. If I was doing this again, I'd favour the more simple notice approach in #52979. For example, it'd probably be possible to scan the user's messages for keywords and then display a notice based on it. 

<img width="1816" alt="Screenshot 2021-05-29 at 07 36 26" src="https://user-images.githubusercontent.com/43215253/120060608-97b3ed00-c050-11eb-892c-0cd9e6db9c1d.png">

It wouldn't be perfect, and sometimes it'd make the wrong recommendations since we'd need to rely on the user comparing their issue to the one in the notice, but it's probably a simpler experience. The main issue regarding Jetpack sites using this form should also be alleviated with #52979 so that provides some more space with options here.

I don't really remember all of my thought process behind this anymore, but to me today, it does feel more complicated than it needs to be. I'm totally up for further discussion if we want to continue going down this road though.

KokkieH commented:
Wow! I'd forgotten about this one 😄 

@sixhours this definitely seems to cover the most common scenarios to get people routed to the best place for support (thanks for making the video walkthrough!), though I'm concerned that it might be more confusing than helpful for most people, and actually end up making it harder to submit a support request. Especially for our Business and eCommerce customers, I'd say we want getting connected to live chat to have as little mental overhead as possible. 

But perhaps we already have that direct line with the inline help icon appearing on all Calypso pages, and perhaps there's room for a more detailed approach on the separate support contact form.

It might be good to get input from someone in HappOps on this as well. cc @mattwondra for input/suggestions who to ping in that regard.

As for the original issues this was meant to address, #16016 and #38280, I think the PR in #52979 mentioned above is a more direct fix, but that doesn't mean there isn't room for improvement to the contact form in general.

Aurorum commented:
@KokkieH What would you think about https://github.com/Automattic/wp-calypso/pull/35944#issuecomment-850781154 where a link is displayed in a notice based on what's been typed in the form? That could also easily be hidden for paid customers, and should be much less confusing. 

> I'm concerned that it might be more confusing than helpful for most people, and actually end up making it harder to submit a support request.

I agree with this - I think that's very likely to end up happening, in all honesty.

KokkieH commented:
> What would you think about #35944 (comment) where a link is displayed in a notice based on what's been typed in the form? That could also easily be hidden for paid customers, and should be much less confusing.

If I'm not mistaken the inline contact form already does something like this, or at least, at one point we were working on it. Matt, who I cc'd above, will know more about that, so I'll leave him to comment on the feasibility of that :)

mattwondra commented:
> It might be good to get input from someone in HappOps on this as well. cc @mattwondra for input/suggestions who to ping in that regard.

@kristenzuck would have better input than me on this.

My general take sounds like it aligns with @Aurorum and @KokkieH — this PR has a solution from 21 months ago to a problem first reported 4 years ago, and contains major changes to support flows that haven't been evaluated in our current context.

@Aurorum — I'm deeply appreciative of the time you spent, but it seems likely that the best course is to abandon this PR and re-evaluate the problem with fresh eyes. If there's a lighter-weight solution, that's almost certainly a better option as a stepping stone.

mattwondra commented:
> If I'm not mistaken the inline contact form already does something like this, or at least, at one point we were working on it.

We have a Q&A system ("Sibyl") that recommends Support docs based on keywords in the user's text ("Does this help?"). But it's not nearly as assertive as the notice I see in that screenshot ("Hey, you might be at the wrong place"). So if that direction was desirable, I'd say it should probably be written as a one-off part of the contact form, directly in Calypso code.

Aurorum commented:
No worries - agreed! 

> I'd say it should probably be written as a one-off part of the contact form, directly in Calypso code.

If people feel there's merit in that idea, feel free to let me know and I'll come up with a PR this week or next. :)

supernovia commented:
I'd totally forgotten this one too. What a year!

Thank you, @Aurorum .
